README.md

```diff
@@ -1,20 +1,246 @@
+dlo-debug.js:1136   [2] insert output (numKeys=0 sizeOfValues=0 sizeOfPayload=34)
+  ['Order Completed',{'transactionID':'2325112','grandTotal':45.59,'subTotal':37.02,'shippingTotal':5.99,'totalTax':2.58,'promotionCodes':[],'selectedPayment':'VISA','deliveryTotal':0,'shippingTax':0,'totalAdjustment':0}]
+dlo-debug.js:1136   [3] suffix output (numKeys=0 sizeOfValues=0 sizeOfPayload=34)
+  ['Order Completed',{'transactionID_str':'2325112','grandTotal_real':45.59,'subTotal_real':37.02,'shippingTotal_real':5.99,'totalTax_real':2.58,'promotionCodes_strs':[],'selectedPayment_str':'VISA','deliveryTotal_int':0,'shippingTax_int':0,'totalAdjustment_int':0}]
+
+Order Completed {transactionID_str: '2325112', grandTotal_real: 45.59, subTotal_real: 37.02, shippingTotal_real: 5.99, totalTax_real: 2.58, …}
+```
+
+In addition to showing the data conversions, debug includes the following statistics.
+
+- The number of properties in the object shown as `numKeys`.
+- The size (assuming UTF16) of the values in the object shown as `sizeOfValues`.
+- The size (assuming UTF16) of the object when it is stringified shown as `sizeOfPayload`.
+
+## Deploying Rules
+
+Once rules have been verified in a test environment, they can be moved to production.  New rules can be added to any existing rules such as those in `window['_dlo_rules']`.  Alternatively, new rules can be placed in their own rule set to allow contribution from different teams or parts of a site.  This can be done by creating a new configuration option that begins with `_dlo_rules` - for example `window['_dlo_rules_cart_and_checkout']`.  When DLO initializes all properties that begin with `_dlo_rules` will be processed.
+
+After new rules are deployed, visit the site and ensure the data is being sent to the destination.  One approach is to simply review in FullStory that the custom event or user variable has been set.  If data is not found, set the `debug` property to `true` for any problematic rules.  If problems still persist, see the Monitoring section to verify something else isn’t at fault.
+
+## Monitoring
+
+By default, DLO will output any logging related messages to the browser’s JavaScript console.  For production environments, a custom log appender is preferable, and it will allow you to redirect logging to a centralized location.
+
+Below is an example of adding a custom FullStory log appender. (Be aware of [rate limiting](https://help.fullstory.com/hc/en-us/articles/360020623234#custom-property-rate-limiting) if sending logs to FullStory; however, DLO only logs when errors occur.)  When a log event is generated by DLO, it will pass the event to the custom appender, which sends the information to FullStory as a custom event.
+
+```javascript
+window['_dlo_appender'] = {
+ log: ({ level: level_int, message: message_str, datalayer: datalayer_str }) => {
```

Comment:
If you destructuring an object, your method signature would look like this:

`log: ({ level, message, datalayer }) => {`

and then the call to FS.event would be:

`FS.event('Data Layer Observer', { level, message, datalayer });`

Reply:
@patrick-fs This uses destructing and aliasing.  The FullStory type suffixes `_int` for example are getting added as part of the aliasing.  This way the object sent to FS.event has the appropriate suffixes.

Reply:
oh https://davidwalsh.name/destructuring-alias

I didn't know that aliasing was a thing. Using aliases to add the suffixes makes sense.